inspirehep/modules/search/walkers/elasticsearch.py

```diff
@@ -90,6 +90,11 @@ def visit(self, node):
     @visitor(Value)
     def visit(self, node):
         def query(keyword):
+            # FIXME: This is a temporary hack that should be removed when
+            # nested keywords search is implemented.
+            if str(node.value).startswith('recid:'):
+                node.value = node.value[len('recid:'):]
+
```

Comment:
Yes of course. But since we have the functionality that enables us to search references based on recids why not use it? 

Also when we introduce second order operations this will just be ignored (until someone sees the fixme tag and actually remove it) since there will be no values starting with `recid:` . So we don't risk braking anything in the future (I hope :smiley: ).